manager/integration/tests/test_csi.py

```diff
@@ -536,3 +536,119 @@ def test_csi_expansion_with_replica_failure(client, core_api, storage_class, pvc
     write_pod_volume_data(core_api, pod_name, test_data)
     resp = read_volume_data(core_api, pod_name)
     assert resp == test_data
+
+@pytest.mark.skip(reason="TODO") # NOQA
+def test_csi_volumesnapshot_basic():  # NOQA
+    """
+    Test creation / restoration / deletion of a backup via the csi snapshotter
+
+    Context:
+
+    We want to allow the user to programmatically create/restore/delete longhorn backups
+    [csi snapshot mechanism](https://kubernetes.io/docs/concepts/storage/volume-snapshots/)
+
+    Setup:
+
+    1. Make sure your cluster contains the below crds
+    https://github.com/kubernetes-csi/external-snapshotter/tree/master/client/config/crd
+    2. Make sure your cluster contains the snapshot controller
+    https://github.com/kubernetes-csi/external-snapshotter/tree/master/deploy/kubernetes/snapshot-controller
+
+    Steps:
+
+    def csi_volumesnapshot_creation_test(snapshotClass=longhorn|custom):
+    1. create volume(1)
+    2. write data to volume(1)
+    3. Create a kubernetes `VolumeSnapshot` object with a test uuid
+    4. The `VolumeSnapshot.uuid` will be used to identify a **longhorn snapshot**
+       and the associated `VolumeSnapshotContent` object
+    5. check creation of a new longhorn snapshot named `snapshot-uuid`
+    6. check for `VolumeSnapshotContent` named `snapcontent-uuid`
+    7. wait for `VolumeSnapshotContent.readyToUse` flag to be set to **true**
+    8. check for backup existance on the backupstore
+
+    # the csi snapshot restore sets the fromBackup field same as the StorageClass based restore approach.
+    def csi_volumesnapshot_restore_test():
+    9. create a `PerstientVolumeClaim` object where the `dataSource` field references
+       the `VolumeSnapshot` object by name
+    10. verify creation of new volume(2)
```

Comment:
Just to confirm, the creation of PVC will create the volume(2)?

Reply:
That is correct, I will reword that.